### DISCUSSION TITLE
Is this a SMARTS bug? Trying to find a "S" atom with at least one hydrogen

### DISCUSSION THREAD

**Discussion by simonmb** [rdkit/rdkit#4598] [Q&A] [answered]
I am trying to find several atoms on a molecule. For the "S" atom I am getting the following:
```
mol = Chem.MolFromSmiles('CC(=N)S')
patt = Chem.MolFromSmarts('[OX2H1,#7H2,#7H1,Sh1]')
mol.GetSubstructMatches(patt)
```
This returns ((2,), (3,))
I would expect only the "N" atom to be found.

**Comment by simonmb** [comment, marked as answer]
Can somebody please delete this discussion? I just saw my mistake. And I am unable to find the link to delte this. Sorry.

**Reply by bp-kelley** [reply]
I marked the question as answered for now.
